tests/scripts/all.sh

```diff
@@ -898,9 +898,12 @@ component_test_full_cmake_clang () {
 }
 
 component_build_deprecated () {
-    msg "build: make, full config + DEPRECATED_WARNING, gcc -O" # ~ 30s
+    msg "build: make, full config - deprecated + DEPRECATED_WARNING, gcc -O" # ~ 30s
```

Comment:
Partially pre-existing: I don't understand the purpose of this component. Probably deserves a comment in the code.

Reply:
In the meantime I saw your reply to Ronald. I think something similar should be added as comments in the code.

The comments could also perhaps mention one point that wasn't immediately clear to me: we need to disable deprecated `config.h` options when building with `DEPRECATED_WARNINGS`  because they use `#warning` which isn't affected by `-Wno-deprecated` (unlike deprecated functions which use the `deprecated` attribute).

Reply:
AFAIR that code has existed since before the componentization of `all.sh`. The pure-`full` jobs might not have!